src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -0,0 +1,430 @@
+
+                    case QUIC_CONNECTION_EVENT.STREAMS_AVAILABLE:
+                        {
+                            status = HandleEventStreamsAvailable(
+                                connectionEvent);
+                        }
+                        break;
+
+                    default:
+                        break;
+                }
+            }
+            catch (Exception)
+            {
+                return MsQuicConstants.InternalError;
+            }
+
+            return status;
+        }
+
+        private uint HandleEventConnected(ConnectionEvent connectionEvent)
+        {
+            if (NetEventSource.IsEnabled) NetEventSource.Enter(this);
+
+            _connectTcs?.SetResult(null);
+            _connectTcs = null;
+
+            if (NetEventSource.IsEnabled) NetEventSource.Exit(this);
+            return MsQuicConstants.Success;
+        }
+
+        private uint HandleEventShutdownBegin(ConnectionEvent connectionEvent)
+        {
+            if (NetEventSource.IsEnabled) NetEventSource.Enter(this);
+
+            if (NetEventSource.IsEnabled) NetEventSource.Info(this, $"Shutdown begin {MsQuicConstants.ErrorTypeFromErrorCode(connectionEvent.ShutdownBeginStatus)}");;
+
+            _connectTcs?.SetResult(null);
+            _connectTcs = null;
+
+            if (NetEventSource.IsEnabled) NetEventSource.Exit(this);
+
+            return MsQuicConstants.Success;
+        }
+
+        private uint HandleEventShutdownBeginPeer(ConnectionEvent connectionEvent)
+        {
+            return MsQuicConstants.Success;
+        }
+
+        private uint HandleEventShutdownComplete(ConnectionEvent connectionEvent)
+        {
+            if (NetEventSource.IsEnabled) NetEventSource.Enter(this);
+
+            if (NetEventSource.IsEnabled) NetEventSource.Info(this, $"Shutdown Complete");
+
+            _shutdownTcs?.SetResult(null);
+
+            if (NetEventSource.IsEnabled) NetEventSource.Exit(this);
+            return MsQuicConstants.Success;
+        }
+
+        private uint HandleEventNewStream(ConnectionEvent connectionEvent)
+        {
+            if (NetEventSource.IsEnabled) NetEventSource.Enter(this);
+
+            MsQuicStream msQuicStream = new MsQuicStream(_api, this, connectionEvent.StreamFlags, connectionEvent.Data.NewStream.Stream, inbound: true);
+
+            _acceptQueue.Writer.TryWrite(msQuicStream);
+
+            if (NetEventSource.IsEnabled) NetEventSource.Exit(this);
+
+            return MsQuicConstants.Success;
+        }
+
+        private uint HandleEventStreamsAvailable(ConnectionEvent connectionEvent)
+        {
+            return MsQuicConstants.Success;
+        }
+
+        internal override async ValueTask<QuicStreamProvider> AcceptStreamAsync(CancellationToken cancellationToken = default)
+        {
+            if (NetEventSource.IsEnabled) NetEventSource.Enter(this);
+
+            if (await _acceptQueue.Reader.WaitToReadAsync(cancellationToken))
```

Comment:
So what happens if you get a new stream, but the app hasn't accepted it yet, and then that stream gets a receive call?

Reply:
The receive call will still be handled as the object is still created. The app doesn't need to call ReadAsync for the receive to be handled.